turbinia/api/utils.py

```diff
@@ -48,11 +69,31 @@ def get_task_output_path(request_id: str, task_id: str) -> str:
       log.error(message)
       raise HTTPException(status_code=404, detail=message) from exception
 
-    for request_dir in request_dirs:
-      if task_id in request_dir:
-        request_output_path = os.path.join(request_output_path, request_dir)
-        break
-    return request_output_path
+    latest_timestamp = 0
+    for task_dir in request_dirs:
+      # We'll check the timestamp that's part of the directory name
+      # to make sure we grab the most recent directory for the task
+      # in case there are more than one.
+      if task_id in task_dir:
+        timestamp = int(task_dir.split('-')[0])
+        if timestamp > latest_timestamp:
+          latest_timestamp = timestamp
+          task_output_path = os.path.join(request_output_path, task_dir)
+
+    return task_output_path
+
+
+def get_plaso_file_path(task_id: str) -> str:
+  """Returns the saved_path for a specific Plaso task."""
+  task = get_task_objects(task_id)[0]
+  output_dir = turbinia_config.toDict().get('OUTPUT_DIR')
+
+  if not task or not task.get('name').startswith('Plaso'):
+    raise TurbiniaException(f'Task {task_id} is invalid for this operation.')
```

Comment:
Tiny nit, but maybe say something about why it's invalid, like "Task {foo} not found or does not  start with 'Plaso'" or something similar

Reply:
Done